sally/models/stock_warehouse_weather.py

```diff
@@ -0,0 +1,84 @@
+from odoo import fields, models, api
+from datetime import datetime
+from odoo.exceptions import ValidationError
+import requests
+
+class FlowerShopWeather(models.Model):
+    _name = 'flower.shop.weather'
+    _description = 'weather checking for plants'
+
+    warehouse_id = fields.Many2one('stock.warehouse')
+    temperature = fields.Float("Temperature")
+    pressure = fields.Float()
+    humidity = fields.Float()
+    wind_speed = fields.Float()
+    rain_volume = fields.Float()
+    description = fields.Char()
+    capture_time = fields.Datetime()
+    
+class warehouse(models.Model):
+    _inherit = 'stock.warehouse'
+
+    weather_ids = fields.One2many('flower.shop.weather', 'warehouse_id')
+    
+    def _get_api_key_and_location(self, show_error):
+        api_key = self.env["ir.config_parameter"].sudo().get_param("flower_shop.weather_api_key")
+        if api_key == "unset" or not api_key:
+            show_error(True)
```

Comment:
The idea behind having this flag `show_error` was to differentiate between a manual button click and a server/scheduled action call. If it is a manual operation where the user is trying to fetch the weather data, errors should be displayed. If it is a scheduled action, then the error message should be logged using a logger.